src/daft-dsl/src/join/mod.rs

```diff
@@ -79,34 +82,3 @@ pub fn infer_join_schema(
         Ok(Schema::new(fields)?.into())
     }
 }
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use crate::col;
-
-    #[test]
-    fn test_get_common_join_keys() {
-        let left_on: &[ExprRef] = &[
-            col("a"),
-            col("b_left"),
-            col("c").alias("c_new"),
-            col("d").alias("d_new"),
-            col("e").add(col("f")),
-        ];
-
-        let right_on: &[ExprRef] = &[
-            col("a"),
-            col("b_right"),
-            col("c"),
-            col("d").alias("d_new"),
-            col("e"),
-        ];
-
-        let common_join_keys = get_common_join_keys(left_on, right_on)
-            .map(|k| k.to_string())
-            .collect::<Vec<_>>();
-
-        assert_eq!(common_join_keys, vec!["a"]);
-    }
-}
```

Comment:
It can be nice to have the tests inline, especially if we want to test private (non-`pub(crate)`) functionality?

Reply:
Hmm... sub modules are still able to view their parent modules private members. The reason why I opted for this is because it just cleans up the file (i.e., keeps the impls in one place and the tests in the other).

I'm not all together too opinionated on this, so let me know if you want this to be changed back.

Reply:
I see people do it both ways. they are functionally the same (inline mod vs non inline)

if one wanted to they could compress an entire crate into one file of nested mod and it would be functionally the same.

Reply:
Ya. I usually prefer having it in a separate file because it reduces visual clutter when navigating through a file.